### ISSUE TITLE
Allow for loops in javascript converter functions 

### ISSUE THREAD

**Issue by aedelmann** [eclipse/vorto#2029]
Motivation: 
https://stackoverflow.com/questions/58411197/parsing-non-fixed-format-binary-payload-with-a-custom-javascript-conversion-in-v 

Confirmations:
- Javascript function evaluator allows the definition of for loops

Current workaround:

Creating a java converter function (as opposed to javascript) and own namespace and reference it from the mapping rules. The java converter function would need to be registered in the mapping engine when creating the data mapper, like 

```
DataMapperBuilder builder =
        IDataMapper.newBuilder().registerScriptEvalProvider(new JavascriptEvalProvider())
            .registerConverterFunction(MyCustomConverterFunction())
           // adding standard functions provided by vorto, e.g. string, number, ... if needed.
            .withSpecification(specification);
```

**Comment by glennergeerts-aloxy** [comment]
Allowing for loops in the javascript engine is a part of the solution indeed.

Besides this, it is also required to be able to passs a bytearray to the javascript function. I did not succeed in this during my testing, but maybe I did something wrong. Being able to pass a bytearray might already enable some more mapping use cases which don't require looping per se.

**Comment by aedelmann** [comment]
fixed in 0.12.3
